org.osgi.service.component/src/org/osgi/service/component/ComponentContext.java

```diff
@@ -49,22 +49,19 @@
 
 	/**
 	 * Returns the service object for the specified reference name.
-	 * 
 	 * <p>
 	 * If the cardinality of the reference is {@code 0..n} or {@code 1..n} and
-	 * multiple services are bound to the reference, the service with the
-	 * highest ranking (as specified in its {@code Constants.SERVICE_RANKING}
-	 * property) is returned. If there is a tie in ranking, the service with the
-	 * lowest service id (as specified in its {@code Constants.SERVICE_ID}
-	 * property); that is, the service that was registered first is returned.
+	 * multiple services are bound to the reference, the service whose
+	 * {@code ServiceReference} is first in the ranking order is returned. See
```

Comment:
previously highest rank returned, now lowest rank? probably the last one is returned.

Reply:
As it now says, it returns the first in ranking order. Ranking order is defined as the reverse of natural order.  So this is the same as before. See <https://docs.osgi.org/specification/osgi.core/8.0.0/framework.service.html#framework.service.servicerankingorder>.